0000-net-mac-address.md

```diff
@@ -0,0 +1,72 @@
+- Feature Name: `net_address_mac`
+- Start Date: 2017-07-26
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+Add a media access control address (MAC) address datatype, `std::net::MacAddr48` to `std::net`.
+
+# Motivation
+[motivation]: #motivation
+
+Currently there is no standard way to communicate physical (or ethernet) addresses when doing network related development in rust, even though `std::net::IpAddrV4` and `std::net::IpAddrV6` exist. The MAC address is, however, a regularly occurring data-structure when doing any type of network related development.
+
+There is also a proliferation of implementations which are not compatible with each other, forcing developers to manually implement the data type again and again, reducing the opportunity for code re-use and convenience. `nom`[1], `libpnet`[2] and `diesel`[3] being a couple of examples.
```

Comment:
This *exact*  argument applies equally to dozens if not hundreds of types. If I took this this RFC and find/replaced `MacAddr48` with `Uuid` or `Path` or `Polygon` or `Rgb` or `Rgba` or `Netmask` or `Quaternion` or `Matrix` or `Vector` or `TraceId` and what would change? Libstd should not be the dumping ground for every possible type that could exist.

The IpAddr types are defined in the standard library because they're used in networking APIs. MacAddr48 on the other hand would be an orphaned type which interacts with nothing.